### ISSUE TITLE
Does this code support multiple GPU cards?

### ISSUE THREAD

**Issue by yfliao** [buriburisuri/speech-to-text-wavenet#24]
Dear All,

Does this code could use multiple GPUs at the same time? I have eight GPUs in my server, but it seems it only used one GPU (although it occupied all GPUs). If I want to take advantage of all my GPUs, how should I modify the code (where to start)? 

Thanks for your help and have a nice day!

Best Regards,
yuanfu

PS: Here is the log provided by "nvidia-smi"
+-----------------------------------------------------------------------------+
Wed Jan  4 15:38:35 2017       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 367.57                 Driver Version: 367.57                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1080    On   | 0000:04:00.0     Off |                  N/A |
| 34%   55C    P2    51W / 198W |   7831MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 1080    On   | 0000:05:00.0     Off |                  N/A |
|  0%   32C    P8    11W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   2  GeForce GTX 1080    On   | 0000:08:00.0     Off |                  N/A |
|  0%   33C    P8    12W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   3  GeForce GTX 1080    On   | 0000:09:00.0     Off |                  N/A |
|  0%   31C    P8    13W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   4  GeForce GTX 1080    On   | 0000:83:00.0     Off |                  N/A |
|  0%   31C    P8    13W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   5  GeForce GTX 1080    On   | 0000:84:00.0     Off |                  N/A |
|  0%   31C    P8    12W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   6  GeForce GTX 1080    On   | 0000:87:00.0     Off |                  N/A |
|  0%   34C    P8    12W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
|   7  GeForce GTX 1080    On   | 0000:88:00.0     Off |                  N/A |
|  0%   32C    P8    12W / 198W |   7715MiB /  8113MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0      3158    C   python                                        7829MiB |
|    1      3158    C   python                                        7713MiB |
|    2      3158    C   python                                        7713MiB |
|    3      3158    C   python                                        7713MiB |
|    4      3158    C   python                                        7713MiB |
|    5      3158    C   python                                        7713MiB |
|    6      3158    C   python                                        7713MiB |
|    7      3158    C   python                                        7713MiB |
+-----------------------------------------------------------------------------+

**Comment by buriburisuri** [comment]
@yfliao Sorry for late reply. ( enjoyed long holidays ).  I guess that your nvidia setting cause some problems.  Easy trick to escape this will be 'CUDA_VISIBLE_DEVICE' env variable as followings:

> CUDA_VISIBLE_DEVICE=0 python train.py 

Thank you.

**Comment by vjdtao** [comment]
@buriburisuri is "CUDA_VISIBLE_DEVICE" a typo? Should it be "CUDA_VISIBLE_DEVICES"?

Does setting "CUDA_VISIBLE_DEVICE=0" mean to force the use of only GPU0? If so, this will not run the all 8 GPUs as @yfliao asked.

**Comment by buriburisuri** [comment]
@vjdtao, @yfliao  Yes, you're right. "CUDA_VISIBLE_DEVICES" is correct. For now, single GPU training is available but I'll add multiple GPUs support.
